tests/fixtures/fixture_portainer.py

```diff
@@ -0,0 +1,54 @@
+import subprocess
+from typing import Literal
+
+import pytest
+import requests
+from tenacity import retry, retry_if_exception_type, stop_after_attempt, wait_random
+from yarl import URL
+
+
+def _run_cmd(cmd: str, **kwargs) -> str:
+    result = subprocess.run(
+        cmd, capture_output=True, check=True, shell=True, encoding="utf-8", **kwargs
```

Comment:
it seems using shell=True is a security issue.

using False is better and then you would bass the cmd as a list instead of a str.

Reply:
done whereever possible, its not possible for the actual shell command provided in the dployment agent config, as sometimes shell internals like pipes etc. are used. And in fact they are in the current setup.

For this reason I refactored all calls to exclude shells, execpt for this one mentioned case.